chore(#976): Add all the things to onboarding

andrablaj opened the pull request:
Move `All The Things`onboarding document content under the Contributor Handbook.

Closes https://github.com/medic/cht-docs/issues/976

andrablaj commented:
FYI @latin-panda I started this PR that I have to get back to before new hires join.

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* We’re offline first, so not every outage calls for immediate action/resolution
+* Engineering work - Archv3 (cht-core 4.0 version)
+
+### Product
+#### [CHT Forum](https://forum.communityhealthtoolkit.org/)
+* We’re trying to keep this active
+* Encourage teammates to post and answer questions there instead of slack when the public might benefit
+* Expecting you to be proactive and support the team with checking forum posts and helping when questions arise
+
+#### Partners
+* Medic-hosted
+* Self-hosted
+* Technical partners
+
+#### Product Development Process
+* At Medic, we follow [Continuous Discovery]({{% ref "continuous-discovery-overview" %}}) as a [Product Development Process]({{% ref "product-development-process" %}})
+  * Trying to follow closely with the [Continuous Discovery Habits](https://www.producttalk.org/2021/05/continuous-discovery-habits/) book by Teresa Torres
+  * [Focused Working Groups]({{% ref "focused-groups" %}}) within the product team to focus on specific groups of users
+    * Allies
+    * Care Teams
+    * Infrastructure
+    * Ecosystem
```
@andrablaj In the other PR you reviewed, I made a separate section for the Product Team. Should we keep it together here, or is it a separate file okay?  

We can polish this PR and merge it to start collaborating; it's okay if it's a list of topics for now; it's already a good start. :)

andrablaj replied:
I think the way you organized the page in the other PR is perfect. We can always merge the pages if we change our minds later.

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+
+### What is the CHT?
+* Why the [CHT (Community Health Toolkit)]({{% ref "/why-the-cht" %}})?
+* The [CHT Core Framework]({{% ref "/core" %}}) - Web application that runs inside an Android app (as a [WebView](https://developer.android.com/reference/android/webkit/WebView))
```
As first timer, this webview link is confusing because it doesn't explain much (it's a class definition), and now we also have PWA

```suggestion
* The [CHT Core Framework]({{% ref "/core" %}}) - App that can be accessed in the [browser](https://docs.communityhealthtoolkit.org/core/overview/architecture/#cht-web-application) or as [PWA](https://docs.communityhealthtoolkit.org/core/overview/pwa/) or [native Android app](https://docs.communityhealthtoolkit.org/core/overview/architecture/#cht-android).
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* Why the [CHT (Community Health Toolkit)]({{% ref "/why-the-cht" %}})?
+* The [CHT Core Framework]({{% ref "/core" %}}) - Web application that runs inside an Android app (as a [WebView](https://developer.android.com/reference/android/webkit/WebView))
+* [Offline-First]({{% ref "/core/overview/offline-first" %}}) - for real
+* [Community](https://communityhealthtoolkit.org/) of people and organizations
```
```suggestion
* [Community](https://communityhealthtoolkit.org/) of people and organizations. This is where we document all the things about the CHT Framework. Bookmark it as you primary source of reference. 
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* Internal Operations
+* External Affairs
+
+### Lifecycle of a [CHT Application]({{% ref "/apps" %}}) being built
```
```suggestion
### Lifecycle of a [CHT Application]({{% ref "/running-programs/#cht-lifecycle" %}}) being built
```
this page has a nice diagram

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+## Now, all the things…
+### People/Team
+
+#### Distributed things 
```
```suggestion
#### Distributed team 
```
Does things here means "team"?

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+
+#### Distributed things 
+* Without some effort, it’s easy for things to feel lonely or isolated
+* Default to asynchronous things
```
```suggestion
* Default to asynchronous communications
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+### People/Team
+
+#### Distributed things 
+* Without some effort, it’s easy for things to feel lonely or isolated
```
Do you have a link about remote work as a collaborator? I think you shared it before, but I can't find it... it was about this

andrablaj replied:
Hmm I don't remember. Was it a Medic link or some external resource, do you remember?

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* Respect teammate timezones (including your own!)
+
+#### Expensify
+* Follow up to make sure things get through 
```
Maybe here explain to reach out to internal ops for guidance, like this we can add more value to this bullet

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+#### Expensify
+* Follow up to make sure things get through 
+
+#### Funding
+* Restricted - Clicktime
+* Unrestricted
```
Some thing like: "Getting our clicktime submitted on time is vital to get source the projects adequately, if necessary take extra steps to remind yourself of submitting these. If you have any question about to fill the reports properly, reach out to the finance team"

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* Unrestricted
+
+#### Travel 
+* Meetups at your own risk are fine
```
"Meetups are a great way to build relations with your team! These are usually planned weeks ahead; if you feel comfortable joining, please do!"

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* A few mandatory calls
+* Organization-wide calls are recorded
+* No recurring meetings on Fridays
+* Retros - MetroRetro
```
```suggestion
* Retrospective sessions
```
What is metroRetro?

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* Organization-wide calls are recorded
+* No recurring meetings on Fridays
+* Retros - MetroRetro
+* 2 Daily Standups
```
```suggestion
* Daily Standups available in 2 timezones.
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* No recurring meetings on Fridays
+* Retros - MetroRetro
+* 2 Daily Standups
+* Weekly FWG meetings
```
```suggestion
* Weekly Focused Working Groups meetings
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+
+### Process
+
+#### Development
```
Maybe link to [this workflow](https://docs.communityhealthtoolkit.org/contribute/code/workflow/)

Also numeration can be replaced by bullets, easier to maintain

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+
+#### Development
+* Basic current process:
+1. Take ticket (GitHub issue from a board)
```
```suggestion
1. Take ticket (GitHub issue from your Focused Working Group's board)
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+1. Take ticket (GitHub issue from a board)
+1. Code something up on a branch
+1. Open pull request
+1. Move ticket to “Ready for AT (Acceptance Testing)”
```
No longer. But can be replaced "Get [quality assistance](https://docs.communityhealthtoolkit.org/contribute/medic/product-development-process/quality-assistance/) from the start"

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* Basic current process:
+1. Take ticket (GitHub issue from a board)
+1. Code something up on a branch
+1. Open pull request
```
```suggestion
1. Open pull request add the QA engineer and another developer as reviewers.
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+1. Move ticket to “Ready for AT (Acceptance Testing)”
+1. Upon testing passing, merge and delete branch
+1. Get QA engineers involved early in the process!
+* Releases
```
https://docs.communityhealthtoolkit.org/contribute/code/releasing/

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+1. Upon testing passing, merge and delete branch
+1. Get QA engineers involved early in the process!
+* Releases
+1. Backwards compatibility matters a lot
```
```suggestion
1. Backwards compatibility matters a lot, so CHWs can keep using the app and delivery care to their community without interruptions. 
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+  * cht-sync 
+  * couch2pg
+  * cht-pipeline
+  * cht-sync
```
repeated

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+  * [Ecosystem Workstream](https://github.com/orgs/medic/projects/134/views/11)
+  * [Allies Workstream](https://github.com/orgs/medic/projects/134/views/3)
+  * [Care Teams Workstream](https://github.com/orgs/medic/projects/134/views/2)
+  * [Test Automation](https://github.com/orgs/medic/projects/134/views/12)
```
Infra FWG: https://github.com/orgs/medic/projects/134/views/26

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+
+#### QA
+* Doing a lot of automation
+* Trying to move from manual AT and release testing to fully automated
+* Trying out [quality assistance]({{% ref "quality-assistance" %}})
```
```suggestion
* We moved from manual AT and release testing to fully automated
* We do [quality assistance]({{% ref "quality-assistance" %}})
```

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,135 @@
+* Ticketing system: only GitHub
+* Not on-call
+* We’re offline first, so not every outage calls for immediate action/resolution
+* Engineering work - Archv3 (cht-core 4.0 version)
```
I think we dont need this last bullet, since we overcame the big stuff from transitioning from 3.x to 4.x

But we can include this https://docs.communityhealthtoolkit.org/core/overview/architecture/ in the section about CHT from top.

latin-panda reviewed (changes requested):
Apologies so the delay on a late proper review. It would be nice to merge this! I've added some suggestions, let me know if you need help :)

latin-panda commented on content/en/contribute/medic/onboarding/all-the-things.md:
Oh it'd be nice to mention the cht academy course as a good way to understand how CHT works.

andrablaj commented:
@latin-panda this PR is so outdated! 🤦‍♀️ Thanks for the fantastic suggestions; I am on it and will share a non-draft PR soon!

andrablaj commented:
@latin-panda thank you again for your suggestions, I incorporated them the this new version. Can you please have another look?

@craig-landry, I added you as a reviewer, as this was inspired by the `All the Things `document you shared with me when I joined Medic. The purpose of this page is to replace that Google doc with a few additions (as some things have changed since I joined 🎆 ). Please let me know if you find something missing from the list.

latin-panda reviewed (approved):
This is good 🤩, from here we can do incremental improvements in separate PRs
Thank you!

andrablaj commented:
@esthermmoturi, I am tagging you here in case you have any suggestions for this PR following our conversation on Slack. You asked why we are adding content here vs. Notion. The reasoning behind this is mostly for the easiness of contributing and discussing the content and also bringing more visibility to our onboarding to the community outside Medic. 
Please let me know if you have further thoughts!

esthermmoturi commented:
Hey @andrablaj , I have looked at the resources you shared and the conversations. I would recommend moving the documentation to a separate public sub domain like handbook.communityhealthtoolkit.org. The reason is that the information that has been added is only on Product and doesn't reflect the wholeness/ presence of other departments. It's also because the objective of docs.communityhealthtoolkit.org is documentation on the product(CHT) and not the people. We can keep the [code section](https://docs.communityhealthtoolkit.org/contribute/code/) and the [docs section](https://docs.communityhealthtoolkit.org/contribute/docs/) as it is directly connected to the product.

craig-landry commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,140 @@
+* The [CHT Core Framework]({{% ref "/core" %}}) - App that can be accessed in the [browser](https://docs.communityhealthtoolkit.org/core/overview/architecture/#cht-web-application) or as [PWA](https://docs.communityhealthtoolkit.org/core/overview/pwa/) or [native Android app](https://docs.communityhealthtoolkit.org/core/overview/architecture/#cht-android).
+* [Offline-First]({{% ref "/core/overview/offline-first" %}}) - for real
+* [Architecture of the CHT]({{% ref "core/overview/architecture" %}})
+* [Community](https://communityhealthtoolkit.org/) of people and organizations. This is where we document all the things about the CHT Framework. Bookmark it as you primary source of reference. 
```
```suggestion
* [Community](https://docs.communityhealthtoolkit.org/) of people and organizations. This is where we document all the things about the CHT Framework. Bookmark it as you primary source of reference. 
```

The text here makes me expect to go to the docs site. Was that the intention?

craig-landry commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,140 @@
+
+#### Travel 
+* Team Meetups are a great way to build relations with your team! These are usually planned weeks ahead; if you feel comfortable joining, please do!
+* Focused Working Groups team members regurarly organize in-person meetups to meet with the people they serve. It's highly recommended to join those trips to get more connected to the team and the mission! 
```
This feels more accurate... and is easier to spell 😉

```suggestion
* Focused Working Groups team members may sometimes organize in-person meetups to meet with the people they serve. It's highly recommended to join those trips to get more connected to the team and the mission! 
```

craig-landry commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,140 @@
+* Organization-wide calls are recorded.
+* No meetings on Fridays, as we consider Fridays as Deep Work days! 
+* Retrospective sessions
+* Daily Standups available in 2 timezones
```
I hadn't read this in a while and doing so now made me think we're asking people to attend both.

```suggestion
* Daily Standups available in 2 timezones (pick the one that suits you best)
```

craig-landry commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,140 @@
+* [Data Flow]({{% ref "core/overview/data-flows-for-analytics" %}})
+* [Monitoring & Alerting]({{% ref "apps/guides/hosting/monitoring" %}})
+* The main repositories to look at:
+  * cht-core
+  * cht-conf
+  * cht-android 
+  * cht-sync 
+  * cht-pipeline
+  * couch2pg
+  * cht-watchdog
```
I just figured I'd link these while in here doing the review
```suggestion
  * [cht-core](https://github.com/medic/cht-core)
  * [cht-conf](https://github.com/medic/cht-conf)
  * [cht-android](https://github.com/medic/cht-android) 
  * [cht-sync](https://github.com/medic/cht-sync) 
  * [cht-pipeline](https://github.com/medic/cht-pipeline)
  * [couch2pg](https://github.com/medic/couch2pg)
  * [cht-watchdog](https://github.com/medic/cht-watchdog)
```

craig-landry commented on content/en/contribute/medic/onboarding/all-the-things.md:
```diff
@@ -0,0 +1,140 @@
+
+### Product
+#### [CHT Forum](https://forum.communityhealthtoolkit.org/)
+* We’re trying to keep the Forum active
```
As we keep doing better and better I think this is now a fair way to word it:

```suggestion
* We keep the forum active. It's a great place to talk with people working with the CHT.
```

craig-landry reviewed (changes requested):
Mostly all good. While in there I took a moment to update a few things.
